src/datasources/test-plans/components/TestPlansVariableQueryEditor.test.tsx

```diff
@@ -33,31 +41,35 @@ describe('TestPlansVariableQueryEditor', () => {
   }
 
   it('should render default query', async () => {
-    const container = renderElement();
+    await act(async () => {
+      renderElement();
+    });
```

Comment:
I wasn't familiar with it but looks like [wrapping component rendering in React with `act` calls](https://react.dev/reference/react/act#rendering-components-in-tests) is an established pattern. And now components have a consistent async `useEffect` to wait for.

Instead of wrapping every call to our custom `renderElement` in an `act` statement can we instead make `renderElement` async and perform the `act` consistently? Seems like it would help with making sure it's used consistently.